### DISCUSSION TITLE
Status "pending" to "complete" FileUpload

### DISCUSSION THREAD

**Discussion by AndreChavs** [primefaces/community#644] [PrimeReact]
How do I display the upload status in the file displayed on the screen and change it from "pending" to "complete" when the uploadHandler event is triggered??
I'm using the itemTemplate and customUpload api

**Comment by AndreChavs** [comment]
"I'm using PrimeReact in my project, and I'm having difficulty with FileUpload. In the advanced mode, I can choose the images, but when I use customUpload, I want to change the 'pending' tag to 'complete'. How can I do that?"

**Comment by mertsincan** [comment]
Sorry for the delayed response!
@gucal @nitrogenous can you please assist with the question?

**Comment by nitrogenous** [comment]
Firstly, thank you for your patience! I understand your point but is there any chance can you share a stackblitz or codepen demo with me? So I can have better understanding of your approach and help you. @AndreChavs

**Reply by ifernisimo** [reply]
I have same issue , I can't attach stackblitz but can provide some code pieces.
The problem is - when you select images they have status pending, when you push upload button they goes to server and here if request was successful it may change status on images to 'successful' but it is not, I suppose that I should return some kind of response to upload handler , but what exactly ?

```
 const uploadHandler = (event) => {
    const isMain = event.options.props.id === "main_image";

    setFieldValue("isMain", isMain); <-- formik
    const imagesData = new FormData();

    for (let i = 0; i < event.files.length; i++) {
      imagesData.append("files", event.files[i]);
    }

    if (lodash_get(values, "tempFolder", false)) {
      imagesData.append("tempFolder", values.tempFolder);
    }

    if (isMain) {
      imagesData.append("isMain", true);
    }

    uploadImages(imagesData); <--- RTK mutation hook 
  };
```
  
Primereact Upload component
```

 <FileUpload
            id="place_images"
            mode="advanced"
            name="files[]"
            accept="image/*"
            maxFileSize={10000000}
            customUpload
            multiple
            uploadHandler={(e) => uploadHandler(e, false)}
            emptyTemplate={
              <p className="m-0">Drag and drop files to here to upload.</p>
            }
            invalidFileSizeMessageDetail="Max file size is 10MB"
          />

```

**Reply by nitrogenous** [reply]
Thank you for your understanding and sharing a demo with me. I will take a deep look at it and fix this as soon as possible!
